python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -133,19 +109,50 @@ cdef class GroupBy:
         from cudf.core.column_accessor import ColumnAccessor
         cdef vector[libcudf_groupby.aggregation_request] c_agg_requests
         cdef Column col
+        cdef Aggregation agg_obj
 
-        aggregations = _drop_unsupported_aggs(values, aggregations)
+        allow_empty = all(len(v) == 0 for v in aggregations.values())
 
+        included_aggregations = defaultdict(list)
+        idx = 0
         for i, (col_name, aggs) in enumerate(aggregations.items()):
             col = values._data[col_name]
+            dtype = col.dtype
+
+            valid_aggregations = (
+                _LIST_AGGS if is_list_dtype(dtype)
+                else _STRING_AGGS if is_string_dtype(dtype)
+                else _CATEGORICAL_AGGS if is_categorical_dtype(dtype)
+                else _STRING_AGGS if is_struct_dtype(dtype)
+                else _INTERVAL_AGGS if is_interval_dtype(dtype)
+                else _DECIMAL_AGGS if is_decimal_dtype(dtype)
+                else "ALL"
+            )
+            if (valid_aggregations is _DECIMAL_AGGS
+                    and rmm._cuda.gpu.runtimeGetVersion() < 11000):
+                raise RuntimeError(
+                    "Decimal aggregations are only supported on CUDA >= 11 "
+                    "due to an nvcc compiler bug."
+                )
+
             c_agg_requests.push_back(
                 move(libcudf_groupby.aggregation_request())
             )
-            c_agg_requests[i].values = col.view()
             for agg in aggs:
-                c_agg_requests[i].aggregations.push_back(
-                    move(make_aggregation(agg))
-                )
+                agg_obj = make_aggregation(agg)
+                if (valid_aggregations == "ALL"
+                        or agg_obj.kind in valid_aggregations):
+                    included_aggregations[col_name].append(agg)
+                    c_agg_requests[idx].aggregations.push_back(
+                        move(agg_obj.c_obj)
+                    )
+            if c_agg_requests[idx].aggregations.empty():
+                c_agg_requests.pop_back()
```

Comment:
Instead of push_back and pop_back, create a temp and push to vector only if temp is non-empty.

Reply:
I ran into issues where Cython would try and create a temporary pointer somewhere in the generated code, and since the agg is of type `unique_ptr[aggregation]` that is an invalid copy construction and fails to compile. I'll see if I can find a way around that.

Reply:
Found the problem. In order to handle exceptions in the C++ function (requested by the `except +` in the pxd file) the Cython-generated C++ creates a temporary variable to hold the `unique_ptr[aggregation_request]` on construction, checks for an error, then assigns it to the correct variable if no error was detected. This necessitates and explicit move on construction of the agg request that wouldn't normally be necessary in C++.